### DISCUSSION TITLE
Generate a json file for each feature

### DISCUSSION THREAD

**Discussion by renatasantos** [badeball/cypress-cucumber-preprocessor#1146] [Q&A]
I've been looking for a while and haven't found anything on the subject, is it possible to generate a json file for each feature executed, before it was possible when using the
```
"cypress-cucumber-preprocessor": {
    "stepDefinitions": "./cypress/integration/steps-definitions/",
    "cucumberJson": {
      "generate": true,
      "outputFolder": "cypress/results/cucumber-json",
      "filePrefix": "",
      "fileSuffix": ".cucumber"
    }
```

Now with the changes according to the documentation to generate the jsons files with the configuration
```
"cypress-cucumber-preprocessor": {
    "nonGlobalStepDefinitions": true,
    "filterSpecs": true,
    "omitFiltered": true,
    "stepDefinitions": [
      "cypress/applicant/api/steps-definitions/**/*",
      "cypress/applicant/api/steps-definitions/*",
      "cypress/applicant/e2e/steps-definitions/**/*",
      "cypress/applicant/e2e/steps-definitions/*",
      "cypress/career-page/e2e/steps-definitions/**/*",
      "cypress/recruitment/api/steps-definitions/**/**/*",
      "cypress/recruitment/api/steps-definitions/**/*",
      "cypress/recruitment/e2e/steps-definitions/**/*",
      "cypress/recruitment/e2e/steps-definitions/*",
      "cypress/shared/steps-definitions/*"
    ],
    "html": {
      "enabled": true,
      "output": "cypress/results/cucumber-html/cucumber-report.html"
    },
    "messages": {
      "enabled": true,
      "output": "cypress/results/cucumber-ndjson/cucumber-report.ndjson"
    },
    "json": {
        "enabled": true,
        "output": "cypress/results/cucumber-json/{specName}.json",
        "formater": "cucumber-json-formatter"
      }  
  }
```
 I'm not able to do this, can anyone help me?

**Comment by badeball** [comment]
There is no native support for this anymore. What is your use case, ie. why do you want this?

**Reply by renatasantos** [reply]
In my case, I have several features distributed across 3 systems, we still don't have a pipeline so execution is manual. To ship the release, we run all the tests manually and if a failure occurs during execution, if I have the jsons for features, I can run just this feature, instead of everything again

**Reply by badeball** [reply]
> if I have the jsons for features, I can run just this feature, instead of everything again

How does having a JSON report for each feature affect your ability to run failing scenarios again?

**Reply by renatasantos** [reply]
If I have 200 resources in the same project and one of them breaks, I need to run all 200 again to generate a report.

**Reply by badeball** [reply]
I assume that with one report per feature file, as you refer to in your original post, you would be able to merge one set of reports with the results of a rerun of failed tests and obtain "the best" of the two.

To me, this is akin to falsifying a report. You're trying to hide the fact that some test is flakey. This is not a use-case that I'm considering supporting.

I recommend that you instead look into [test retries](https://docs.cypress.io/guides/guides/test-retries). If a test is retried and succeeds, this will correctly be reflected in any of the report types.
